float64[]  curr_pose
float64[]  tool_coor
float64[]  base_coor